NetworkingComparison/RxSwift/RxSwiftAPI.swift

```diff
@@ -0,0 +1,121 @@
+                var components = URLComponents(url: pathUrl, resolvingAgainstBaseURL: true)
+                else { return nil }
+
+            components.queryItems = API.params.map { URLQueryItem(name: $0, value: $1) }
+            guard let completeUrl = components.url else { return nil }
+            return URLRequest(url: completeUrl)
+        }()
+        guard let request = urlRequest else {
+            return Observable.error(RxSwiftError.badUrl)
+        }
+        return session.rx.data(request: request)
+    }
+
+    private func fetchWithObservableRequest() -> Observable<Data> {
+        print("********** RxSwift requesting with observable URLRequest **********")
+        defer { sessionStyle = .manualResponse }
+
+        // This is somewhat contrived. You can simply build a URLRequest the "standard" way
+        // since the request doesn't change over time.
+        let request: Observable<URLRequest> = Observable.create { observer in
+            guard
+                let pathUrl = URL(string: API.baseUrl)?.appendingPathComponent("forecast"),
+                var components = URLComponents(url: pathUrl, resolvingAgainstBaseURL: true)
+                else {
+                    observer.onError(RxSwiftError.badUrl)
+                    return Disposables.create()
+            }
+            components.queryItems = API.params.map { URLQueryItem(name: $0, value: $1) }
+            guard let completeUrl = components.url else {
+                observer.onError(RxSwiftError.badUrl)
+                return Disposables.create()
+            }
+            observer.onNext(URLRequest(url: completeUrl))
+            observer.onCompleted()
+
+            return Disposables.create()
+        }
+        return request.flatMap(session.rx.data(request:))
+    }
+
+    private func fetchWithManualResponse() -> Observable<Data> {
+        print("********** RxSwift requesting with manual response handling **********")
+        defer { sessionStyle = .standardRequest }
+
+        let urlRequest: URLRequest? = {
+            guard
+                let pathUrl = URL(string: API.baseUrl)?.appendingPathComponent("forecast"),
+                var components = URLComponents(url: pathUrl, resolvingAgainstBaseURL: true)
+                else { return nil }
+
+            components.queryItems = API.params.map { URLQueryItem(name: $0, value: $1) }
+            guard let completeUrl = components.url else { return nil }
+            return URLRequest(url: completeUrl)
+        }()
+        guard let request = urlRequest else {
+            return Observable.error(RxSwiftError.badUrl)
+        }
+        return session.rx.response(request: request).map { response, data in
+            switch response.statusCode {
+            case 200 ..< 300:
+                return data
+            case 401:
+                throw RxSwiftError.invalidKey
+            case 400 ..< 500:
+                throw RxSwiftError.cityNotFound
+            default:
```

Comment:
Nit: NBD, but you're inconsistent about your switch cases new line conventions.

Setting up [SwiftLint](https://github.com/realm/SwiftLint) couldn't hurt! heh

Reply:
I'm personally fluid with this. I just go for clarity. Sometimes it's clear enough on a single line, and saves space. Other times it's more clear with a new line. Other times it's more clear with whitespace between cases.

Here, the `throw`s get offset inconsistently due to the cases being different lengths, so I used newline. Whereas above, they are offset consistently, so single line read cleanly to me. Just personal preference, although of course I'd conform to a team / project's standards where applicable